### ISSUE TITLE
journalbeat-HEAD crashes with interface conversion: interface is string, not common.Time

### ISSUE THREAD

**Issue by basvandijk** [mheese/journalbeat#72]
I noticed that the latest journalbeat release (5.1.2) doesn't work with the latest elasticsearch (5.4). So I upgraded journalbeat to the latest HEAD (7490ae94018b97fe73196c0417a630a4f3393df4). Unfortunately journalbeat now crashes with the following message:

```
May 24 17:03:18 hera systemd[1]: Starting Journalbeat log shipper...
May 24 17:03:18 hera systemd[1]: Started Journalbeat log shipper.
May 24 17:03:18 hera journalbeat[2202]: panic: interface conversion: interface is string, not common.Time
May 24 17:03:18 hera journalbeat[2202]: goroutine 31 [running]:
May 24 17:03:18 hera journalbeat[2202]: panic(0x8ee800, 0xc4208a2140)
May 24 17:03:18 hera journalbeat[2202]:         /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-go-1.7.4/share/go/src/runtime/panic.go:500 +0x1a1
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.getIndex(0xc42028a1e0, 0xe1efa0, 0xc420169680, 0x0, 0x0)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:367 +0xb7
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.createEventBulkMeta(0xe1efa0, 0xc420169680, 0x0, 0xc42028a1e0, 0x0, 0x0, 0x0)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:322 +0xbd
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.bulkEncodePublishRequest(0x7f5280006000, 0xc42022d240, 0xe1efa0, 0xc420169680, 0x0, 0xc420174380, 0x32, 0x32, 0x8bc5c0, 0xc420278db0, ...)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:289 +0xc5
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.(*Client).PublishEvents(0xc4201843c0, 0xc420174380, 0x32, 0x32, 0x0, 0x1, 0x0, 0x0, 0x42f3c3)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:241 +0x13b
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).PublishEvents.func1(0xc4202349c0)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:89 +0x7a
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).publish(0xc4202349c0, 0xe25020, 0xc420356000, 0xc4200e1401, 0xc420853ad8, 0x0, 0x0)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:144 +0x2aa
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).PublishEvents(0xc4202349c0, 0xe25020, 0xc420356000, 0x1, 0xc420174380, 0x32, 0x32, 0x1, 0x1)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:99 +0x74
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.(*elasticsearchOutput).BulkPublish(0xc42016ac00, 0xe25020, 0xc420356000, 0x1, 0xc420174380, 0x32, 0x32, 0xc42003ff88, 0xc420152820)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/output.go:415 +0x79
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).sendBulk(0xc42015e960, 0xc420853c88, 0xc420174380, 0x32, 0x32)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:112 +0x125
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).onBulk(0xc42015e960, 0xc420853c88, 0xc420174380, 0x32, 0x32)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:88 +0x259
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).onMessage(0xc42015e960, 0x0, 0x1, 0xe25020, 0xc420356000, 0x0, 0x0, 0xc420174380, 0x32, 0x32)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:70 +0x9a
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).onEvent(0xc42015e960, 0x0, 0x1, 0xe25020, 0xc420356000, 0x0, 0x0, 0xc420174380, 0x32, 0x32)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:83 +0x74
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).run(0xc42015e960)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:69 +0x290
May 24 17:03:18 hera journalbeat[2202]: created by github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).init
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:57 +0x103
May 24 17:03:18 hera systemd[1]: journalbeat.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
May 24 17:03:18 hera systemd[1]: journalbeat.service: Unit entered failed state.
May 24 17:03:18 hera systemd[1]: journalbeat.service: Failed with result 'exit-code'.
```

This looks very similar to: https://github.com/elastic/beats/issues/1378.

**Comment by nicorevin** [comment]
I was just about to ask whether you overwrite the `@timestamp` or not. 

Because it is explicitly [converted to `common.Time`](https://github.com/mheese/journalbeat/blob/master/beater/journalbeat.go#L157) and can only be overwritten somewhere in libbeat's gut

**Comment by basvandijk** [comment]
Hi @nicorevin, I don't think I'm overwriting `@timestamp`. How should one do that?

It is listed in my template but I don't think that has anything to do with it:

```json
            "@timestamp" : {
               "format" : "strict_date_optional_time||epoch_millis",
               "type" : "date"
            }
```

**Comment by nicorevin** [comment]
This issue can be reproduced if I put 
```yaml
fields:
  "@timestamp": lol
fields_under_root: true
```

into my config. But since I am using logstash for output I can't reproduce it fully. Running `./journalbeat -c ./journalbeat.yml -N -e -d '*'` shows up clearly:
```json
2017/05/25 12:32:18.079336 client.go:214: DBG  Publish: {
  "@timestamp": "lol",
...
```

I don't see any other ways to run into this issue, unless systemd journal will send you the field called `_@TIMESTAMP` or something which I doubt can happen on its own.

**Comment by basvandijk** [comment]
FYI: I've now switched to the logstash output so I'm not having this issue anymore.

**Comment by nicorevin** [comment]
I find this issue very strange. If you could give a way to reproduce this issue I would appreciate!

**Comment by johnmarcou** [comment]
I notice the same issue sometimes, from journalbeat to ES only. I am using ingest pipelines to process data before index, so I keep using JB direct to ES for now. I didn't find the pattern for now, I'll let you know.

**Comment by sbadia** [comment]
Hi!

Same issue here but indeed it's a bit strange, this issue happen only on some nodes of my cluster (with the same configuration on all nodes).

```
# ./journalbeat-debian -c /etc/journalbeat.yml -path.data /var/lib/journalbeat/ -path.home /var/lib/journalbeat/ -e
2017/07/04 14:46:31.484255 beat.go:285: INFO Home path: [/var/lib/journalbeat/] Config path: [/var/lib/journalbeat/] Data path: [/var/lib/journalbeat/] Logs path: [/var/lib/journalbeat//logs]
2017/07/04 14:46:31.484611 beat.go:186: INFO Setup Beat: journalbeat; Version: 5.4.1
2017/07/04 14:46:31.484419 metrics.go:23: INFO Metrics logging every 1m0s
2017/07/04 14:46:31.509758 add_cloud_metadata.go:231: INFO add_cloud_metadata: hosting provider type not detected.
2017/07/04 14:46:31.511935 client.go:123: INFO Elasticsearch url: http://schinf-c2.domain.net:9200
2017/07/04 14:46:31.512217 outputs.go:108: INFO Activated elasticsearch as output plugin.
2017/07/04 14:46:31.513258 publish.go:295: INFO Publisher name: journalbeat
2017/07/04 14:46:31.514000 async.go:63: INFO Flush Interval set to: 1s
2017/07/04 14:46:31.514213 async.go:64: INFO Max Bulk Size set to: 50
2017/07/04 14:46:31.518018 journalbeat.go:50: INFO Seek to tail successful
2017/07/04 14:46:31.521197 beat.go:221: INFO journalbeat start running.
2017/07/04 14:46:31.521245 journalbeat.go:129: INFO Journalbeat is running!
2017/07/04 14:46:31.547858 state.go:130: INFO Loaded 35 events, trying to publish
2017/07/04 14:46:32.247858 client.go:658: INFO Connected to Elasticsearch version 5.4.2
panic: interface conversion: interface {} is string, not common.Time

goroutine 52 [running]:
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.getIndex(0xc420272300, 0x147ec80, 0xc42006b200, 0xc42018d620, 0xb)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:367 +0x57e
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.createEventBulkMeta(0x147ec80, 0xc42006b200, 0xc42018d570, 0xc420272300, 0x0, 0x0, 0x0)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:339 +0x2bf
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.bulkEncodePublishRequest(0x7f497b68a120, 0xc420078198, 0x147ec80, 0xc42006b200, 0xc42018d570, 0xc42006f180, 0x32, 0x32, 0xc8, 0x75d3c2, ...)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:289 +0xeb
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.(*Client).PublishEvents(0xc4200de780, 0xc42006f180, 0x32, 0x32, 0xc3afc8, 0xc4200de780, 0x14f46b0400, 0x0, 0x0)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:241 +0x1a7
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).PublishEvents.func1(0xc420193cb0)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:89 +0x100
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).publish(0xc420193cb0, 0x1485440, 0xc4203710c0, 0x75d301, 0xc42009bb28, 0x14af008, 0x0)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:144 +0x402
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).PublishEvents(0xc420193cb0, 0x1485440, 0xc4203710c0, 0xf40501, 0xc42006f180, 0x32, 0x32, 0xc420219f60, 0x1)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:99 +0xcd
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.(*elasticsearchOutput).BulkPublish(0xc420074e00, 0x1485440, 0xc4203710c0, 0x1, 0xc42006f180, 0x32, 0x32, 0xc42003afb8, 0xc4201769c0)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/output.go:415 +0x9f
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).sendBulk(0xc42007b310, 0xc42003ad38, 0xc42006f180, 0x32, 0x32)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:112 +0x1cb
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).onBulk(0xc42007b310, 0xc42003ad38, 0xc42006f180, 0x32, 0x32)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:88 +0xbf
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).onMessage(0xc42007b310, 0x0, 0x1, 0x1485440, 0xc4203710c0, 0x0, 0x0, 0xc42006f180, 0x32, 0x32)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:70 +0xad
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).onEvent(0xc42007b310, 0x0, 0x1, 0x1485440, 0xc4203710c0, 0x0, 0x0, 0xc42006f180, 0x32, 0x32)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:83 +0xa8
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).run(0xc42007b310)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:69 +0x2f7
created by github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).init
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:57 +0x17d
```

With elasticsearch version 5.4.2 and this config for journalbeat:

```yaml
 ---
name: journalbeat
tags: [ "int2", "journalbeat" ]
logging.level: "info"
logging.to_files: false
logging.to_syslog: true
logging.metrics.enabled: true
logging.metrics.period: 60s
input:
  seek_position: cursor
  cursor_seek_fallback: tail
  write_cursor_state: true
  cursor_state_file: /var/lib/journalbeat/cursor.state
  flush_cursor_secs: 3
  clean_field_names: true
  convert_to_numbers: false
  move_metadata_to_field: journal
processors:
- add_cloud_metadata:
output:
  elasticsearch:
    enabled: true
    hosts: ['schinf-c2.domain.net:9200']
    index: "filebeat-%{+yyyy.MM.dd}"
    pipeline: "journalbeat"
    template.enabled: false
    template.versions.2x.enabled: false
```
I don't modify the `@timestamp` and I also didn't found why on certain nodes it's works well (same bin/systemd version…)

**Comment by samyatchmenoff** [comment]
I started getting the same panic message and it turned out that it was caused by runing journalbeat with a **working directory on a tmpfs partition**. Perhaps this is a useful lead.

**Comment by mheese** [comment]
Smells like the actual issue is in `libbeat` ... Let's see if we are able to reproduce it now.

**Comment by basvandijk** [comment]
Interesting. However in my case I don't specify a `WorkingDirectory` which means [systemd defaults to /](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#WorkingDirectory=).

```
# systemctl cat journalbeat
# /nix/store/d821i03yjaak5z8f0h2fznkimww8aadp-unit-journalbeat.service/journalbeat.service
[Unit]
Description=Journalbeat log shipper

[Service]
Environment="LOCALE_ARCHIVE=/nix/store/5zy82yd0p3scpkp4lmxrz6vf3f9gk1nb-glibc-locales-2.25/lib/locale/locale-archive"
Environment="PATH=/nix/store/c07gdr6cm43j1cphadzafq185k711vx4-coreutils-8.26/bin:/nix/store/7pyzxi7k5l6nym972gi2nq8s9f9b2q0j-findutils-4.6.0/bin:/nix/store/r6s8rcd28wsk4gwviyc93343bq5zwlqq-gnugrep-3.0/bin:/nix/store/xk38vw7z7bfr8173vdwfrfamxqcaj7hi-gnused-4.4/bin:/nix/store/xckqiwxg8la4p7ln22z7ljalbq1x70xn-systemd-232/bin:/nix/store/c07gdr6cm43j1cphadzafq185k711vx4-coreutils-8.26/sbin:/nix/store/7pyzxi7k5l6nym972gi2nq8s9f9b2q0j-findutils-4.6.0/sbin:/nix/store/r6s8rcd28wsk4gwviyc93343bq5zwlqq-gnugrep-3.0/sbin:/nix/store/xk38vw7z7bfr8173vdwfrfamxqcaj7hi-gnused-4.4/sbin:/nix/store/xckqiwxg8la4p7ln22z7ljalbq1x70xn-systemd-232/sbin"
Environment="TZDIR=/nix/store/6cpggd50v2aq5zczn1jjiaavbz3hijf0-tzdata-2016j/share/zoneinfo"

ExecStart=/nix/store/02jx9nk01vhjva4k67lr96b053s0dins-journalbeat-5.1.2-bin/bin/journalbeat -c /nix/store/lwwfd5ncnn078w32baq3fh6p025ia829-journalbeat.yml -path.data /var/lib/journalbeat/data -path.logs /var/lib/journalbeat/logs
ExecStartPre=/nix/store/61gsda4yi7albadi09036c3wx6mi2n1l-unit-script/bin/journalbeat-pre-start
```

**Comment by tylerjl** [comment]
I wanted to add another data point here: I had a journalbeat service running fine, then restarted the service - with no change in the journalbeat binary - and started getting these errors.

Deleting the `.journalbeat-cursor-state` and `.journalbeat-pending-queue` files in the data directory solved this for me - I suspect that the related `tmpfs` comment here may be related, since I assume changing the working directory also forced journalbeat to write new cursor/queue state files.

My (uneducated) hunch is that abruptly stopping the process may have the potential to write bad data into one of these files, at which point the outgoing queue for ES just chokes. I'd be interested to hear if this solves the issue for anyone else.

Great work by the way @mheese, this is an extremely useful beat!

**Comment by CtrlC-Root** [comment]
I also encountered this issue and solved it by setting `WorkingDirectory` in my systemd service file. In case anyone is curious here's what I use: https://github.com/CtrlC-Root/pkg-journalbeat/blob/master/journalbeat.service

**Comment by brucemartins** [comment]
I am seeing this issue as well with the latest release of journalbeat 5.5.2 and ES 5.5.1

**Comment by nicorevin** [comment]
Please try out the latest v5.6.0 version. It is unlikely to fix the issue but should help a bit with narrowing  it down. Also have a look at the linked libbeat issue. ⬆️

**Comment by brucemartins** [comment]
They did change that specific code in 5.6 .. I will see if I can try it out with 5.6 and see if it helps

**Comment by nicorevin** [comment]
They want to narrow down the problem coz it is not obvious where it comes from.

**Comment by jamessewell** [comment]
I just found I can trigger with an incorrect config file like this:

```
journalbeat:
cursor_state_file: "/hab/svc/journalbeat/data/.cursor-state"
pending_queue.file: "/hab/svc/journalbeat/data/.pending-queue"
```

This results in my pending queue being created at `/`:

```
[root@localhost ~]# ls -la /.journalbeat-pending-queue
-rw-------. 1 root hab 1181 Oct  4 10:42 /.journalbeat-pending-queue
```

Which results in the error above till I remove that file.

Changing the config to:

```
journalbeat:
  cursor_state_file: "/hab/svc/journalbeat/data/.cursor-state"
  pending_queue.file: "/hab/svc/journalbeat/data/.pending-queue"
```

Solves the issue. I guess it must be permission based - although I didn't get any more time to play with it.

**Comment by drothlis** [comment]
I'm seeing this issue too with journalbeat 5.6.0 & systemd 229 on Ubuntu 16.04.

I see this log line over & over (formatted for readability and with most fields redacted):

```
client.go:378: ERR @timestamp is not a common.Time but is string in event common.MapStr{
    ...
    "@realtime_timestamp":1.508426556188726e+15,
    "@timestamp":"2017-10-19T15:22:36.188Z",
    "type":"journal",
    "beat":common.MapStr{
        "version":"5.6.0",
        "name":"...",
        "hostname":"..."},
    }
```

I'm running journalbeat without any template (`template.enabled: false`).

**Comment by drothlis** [comment]
Nevermind this seems to be a separate issue.

**Comment by alexjurkiewicz** [comment]
@drothlis I'm running into the same error you are (`@timestamp is not a common.Time but is string in event`) -- did you solve it?

**Comment by petracvv** [comment]
@mheese @nicorevin I found this issue and I think I have a reproducible test case.

I created a Vagrantfile (https://gist.github.com/petracvv/0c8c01563d870e31bc66a7c66fdb7fd3) with which I can reproduce this error.

To reproduce:

1. Download my Vagrantfile and start the VM with vagrant up. I have set up a provisioning script in here that will install the newest elasticsearch 5.6 release via rpm and the newest journalbeat version in `/opt/journalbeat`. The provisioner will also start both services.
2. Once both elasticsearch and journalbeat are running, stop and start the journalbeat service with:
```
[root@journalbeat-test journalbeat]# systemctl stop journalbeat
[root@journalbeat-test journalbeat]# systemctl start journalbeat
```
Once the service restarts the error messages start appearing in the `/opt/journalbeat/logs/journalbeat` log file.

e.g.
```
2018-01-23T20:06:29Z ERR @timestamp is not a common.Time but is string in event common.MapStr{"@realtime_timestamp":1.516737980486525e+15, "boot_id":"d7e2753d58944d1fb22730872230a0df", "type":"journal", "catalog_entry":"Subject: Unit journalbeat.service has begun shutting down\nDefined-By: systemd\nSupport: http://lists.freedesktop.org/mailman/listinfo/systemd-devel\n\nUnit journalbeat.service has begun shutting down.\n", "beat":common.MapStr{"name":"journalbeat-test", "hostname":"journalbeat-test", "version":"5.6.0"}, "cmdline":"/usr/lib/systemd/systemd --switched-root --system --deserialize 21", "code_function":"unit_status_log_starting_stopping_reloading", "systemd_cgroup":"/", "machine_id":"aae3cd63a0be4785a26fd3e931896c2b", "message":"Stopping JournalBeat service...", "exe":"/usr/lib/systemd/systemd", "@timestamp":"2018-01-23T20:06:20.486Z", "pid":"1", "comm":"systemd", "source_realtime_timestamp":"1516737980483849", "gid":"0", "priority":"6", "code_file":"src/core/unit.c", "code_line":"1417", "unit":"journalbeat.service", "transport":"journal", "cap_effective":"1fffffffff", "hostname":"journalbeat-test", "syslog_identifier":"systemd", "message_id":"de5b426a63be47a7b6ac3eaac82e2f6f", "uid":"0", "syslog_facility":"3"}
2018-01-23T20:06:33Z ERR @timestamp is not a common.Time but is string in event common.MapStr{"systemd_cgroup":"/system.slice/polkit.service", "syslog_pid":"625", "priority":"5", "beat":common.MapStr{"name":"journalbeat-test", "hostname":"journalbeat-test", "version":"5.6.0"}, "uid":"999", "gid":"997", "@timestamp":"2018-01-23T20:06:20.482Z", "@realtime_timestamp":1.516737980482641e+15, "systemd_slice":"system.slice", "syslog_identifier":"polkitd", "cap_effective":"0", "source_realtime_timestamp":"1516737980481995", "transport":"syslog", "cmdline":"/usr/lib/polkit-1/polkitd --no-debug", "comm":"polkitd", "machine_id":"aae3cd63a0be4785a26fd3e931896c2b", "syslog_facility":"10", "hostname":"journalbeat-test", "boot_id":"d7e2753d58944d1fb22730872230a0df", "message":"Registered Authentication Agent for unix-process:13763:10628 (system bus name :1.50 [/usr/bin/pkttyagent --notify-fd 5 --fallback], object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-8)", "systemd_unit":"polkit.service", "type":"journal", "exe":"/usr/lib/polkit-1/polkitd", "pid":"625"}
2018-01-23T20:06:33Z ERR @timestamp is not a common.Time but is string in event common.MapStr{"code_file":"src/core/unit.c", "source_realtime_timestamp":"1516737980483849", "gid":"0", "priority":"6", "transport":"journal", "code_line":"1417", "unit":"journalbeat.service", "message_id":"de5b426a63be47a7b6ac3eaac82e2f6f", "uid":"0", "syslog_facility":"3", "cap_effective":"1fffffffff", "hostname":"journalbeat-test", "syslog_identifier":"systemd", "catalog_entry":"Subject: Unit journalbeat.service has begun shutting down\nDefined-By: systemd\nSupport: http://lists.freedesktop.org/mailman/listinfo/systemd-devel\n\nUnit journalbeat.servicehas begun shutting down.\n", "beat":common.MapStr{"hostname":"journalbeat-test", "version":"5.6.0", "name":"journalbeat-test"}, "cmdline":"/usr/lib/systemd/systemd --switched-root --system --deserialize 21", "@realtime_timestamp":1.516737980486525e+15, "boot_id":"d7e2753d58944d1fb22730872230a0df", "type":"journal", "machine_id":"aae3cd63a0be4785a26fd3e931896c2b", "code_function":"unit_status_log_starting_stopping_reloading", "systemd_cgroup":"/", "message":"Stopping JournalBeat service...", "comm":"systemd", "exe":"/usr/lib/systemd/systemd", "@timestamp":"2018-01-23T20:06:20.486Z", "pid":"1"}
```

Hope this helps you track it down. Unfortunately I'm not that familiar with Go to help track this down myself.

**Comment by petracvv** [comment]
I believe I have a fix in Pull #137
